File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cs
using System;

public class Solution
{
    public int solution(int[] nums)
    {
        int answer = 0;
        int n = nums.Length;

        for (int i = 0; i < n - 2; i++)
        {
            for (int j = i + 1; j < n - 1; j++)
            {
                for (int k = j + 1; k < n; k++)
                {
                    int sum = nums[i] + nums[j] + nums[k];
                    if (IsPrime(sum))
                    {
                        answer++;
                    }
                }
            }
        }

        return answer;
    }

    private bool IsPrime(int num)
    {
        if (num < 2) return false;
        for (int i = 2; i * i <= num; i++)
        {
            if (num % i == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<int> nums) 
{
    auto sumThreeElements = [](const vector<int>& nums, int a, int b, int c) 
    {
        return nums[a] + nums[b] + nums[c];
    };

    int count = 0;
    // 모든 조합을 위한 중첩 반복문
    for (int i = 0; i < nums.size(); i++) 
    {
        for (int j = i + 1; j < nums.size(); j++) 
        {
            for (int k = j + 1; k < nums.size(); k++) 
            {
                int sum = sumThreeElements(nums, i, j, k); // 람다 호출
                
                bool isBool = false;
                for(int i=2; i<sum; i++)
                {
                    if(sum%i == 0)
                    {
                        isBool = true;
                    }
                }
                if(!isBool)
                {
                    count++;
                }
            }
        }
    }
    return count;
}